pkg/datapath/linux/config.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2019 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package linux
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"reflect"
+
+	"github.com/cilium/cilium/common"
+	"github.com/cilium/cilium/pkg/byteorder"
+	"github.com/cilium/cilium/pkg/datapath"
+	"github.com/cilium/cilium/pkg/identity"
+	"github.com/cilium/cilium/pkg/labels"
+	"github.com/cilium/cilium/pkg/maps/eppolicymap"
+	ipcachemap "github.com/cilium/cilium/pkg/maps/ipcache"
+	"github.com/cilium/cilium/pkg/maps/lbmap"
+	"github.com/cilium/cilium/pkg/maps/lxcmap"
+	"github.com/cilium/cilium/pkg/maps/metricsmap"
+	"github.com/cilium/cilium/pkg/maps/policymap"
+	"github.com/cilium/cilium/pkg/maps/proxymap"
+	"github.com/cilium/cilium/pkg/maps/sockmap"
+	"github.com/cilium/cilium/pkg/maps/tunnel"
+	"github.com/cilium/cilium/pkg/node"
+	"github.com/cilium/cilium/pkg/option"
+)
+
+// WriteNodeConfig writes the local node configuration to the specified writer.
+func (l *linuxDatapath) WriteNodeConfig(w io.Writer, cfg *datapath.LocalNodeConfiguration) error {
+	fw := bufio.NewWriter(w)
```

Comment:
Open to suggestions whether we should do this buffering inside the datapath implementation outside. I guess strictly speaking we don't need it here because this will be executed to write into a hash object in future, in which case we don't need additional in-memory buffering. In that case I should move it back out to the caller.

The flip side is that we're not checking write error codes for any of the `fmt.Fprintf()` functions below, so we're not able to properly return an error if we don't have a buffer here. That checking would be up to the caller. Of course, we're also relying on the default buffer size for bufio because any writes beyond that size could fail and we won't detect such failures and break out from this function..

Reply:
FWIW the included unit tests use a failing writer to specifically test this case, which makes me feel a bit better about this approach.

Reply:
I'm fine with this as in this PR